### ISSUE TITLE
Generate Tracking docs from JSDoc

### ISSUE THREAD

**Issue by tomalec** [woocommerce/google-listings-and-ads#117]
## TL;DR

Let's generate [Usage Tracking docs](https://github.com/woocommerce/google-listings-and-ads/blob/trunk/src/Tracking/README.md) automatically from code.

## Context

During my onboarding and diving into the project, what bothered me a lot was the lack of outdated docs for many areas in Woo and WP. I got a feeling @ecgan could say a word here as well ;)

I hope we can all agree on the values and benefits of good docs: 
- faster and easier development, as we know what, how, and when to use,
- better quality, as we use the code in the way it was intended to be used,
- faster code reviews, as we know what to expect,
- faster onboarding, as there is less lore to be revealed by elders, and more things is close to the metal/code,
- etc.

The reason for lacking or out-dated doc could be very simple and understandable: "Our team/company has a limited number of people, the project and the code-base are huge, and the documentation is expensive and hard to maintain".

Then I see only one way out: let's make docs easier and cheaper to maintain.

But not to roam too far on high-level dreams, let's focus on a small step we can take as a team: track events.
Hopefully, the solution for it could be extrapolated for the other parts, but we need to start and focus somewhere.

### Current state

1. We record/fire tracking events in [the number of places in code](https://github.com/woocommerce/google-listings-and-ads/search?q=recordEvent) - probably this number will increase with time.
2. We manually document what events and with what parameters are fired in [/src/Tracking/README.md](https://github.com/woocommerce/google-listings-and-ads/blob/trunk/src/Tracking/README.md).
3. Once we change something (a parameter, timing, name) in code, we need to remember to update the README.md.
4. We also need to update the Google Doc.

In my opinion, it's prone to human error, getting out of sync, and simply requires redundant work.

### Proposition

1. Document it in code.
   We can agree on a format later. In principles it would benefit:
   - the developer creating/updating it will need less context-switching. They will implement it in code, and document a line above, without leaving IDE.
   - the developer will get better DevX, as IDE could use that information. Most IDEs support JSDoc pretty well and could use `@fires, @property` and Markdown descriptions, to support development.
2. Generate docs automatically.
   (like on `pre-commit`)
   - less chance for human error,
   - less chance for getting out of sync,
   - more data, we could easily link to the places in code, when a given event is tracked,
   - less human labor :tada: cheaper docs!
   
Cons:

1. We will have to implement JSDoc processing - but hopefully, that would open the door for more future benefits.
2. Syncing with the spreadsheet would still have to be done manually.

#### Format
```js
/**
 * Report table was sorted.
 * @event gla_table_sort
 * @type {Tracking}
 *
 * @property {String} report Id of the report page. E.g. `'dashboard' | 'programs' | 'product-feed'`.
 * @property {String} column Column key.
 * @property {String} direction Direction, `asc` or `desc`.
 */

// then the place/function/component that fires it

/**
 * ...
 * @fires gla_table_sort
 */
```

**Comment by tomalec** [comment]
Closing as done in https://github.com/woocommerce/google-listings-and-ads/pull/1369 and https://github.com/woocommerce/google-listings-and-ads/pull/1401
